simapp/upgrades.go

```diff
@@ -1,38 +1,45 @@
 package simapp
 
 import (
+	"github.com/cosmos/cosmos-sdk/baseapp"
 	storetypes "github.com/cosmos/cosmos-sdk/store/types"
 	sdk "github.com/cosmos/cosmos-sdk/types"
 	"github.com/cosmos/cosmos-sdk/types/module"
-	"github.com/cosmos/cosmos-sdk/x/group"
-	"github.com/cosmos/cosmos-sdk/x/nft"
+	paramstypes "github.com/cosmos/cosmos-sdk/x/params/types"
 	upgradetypes "github.com/cosmos/cosmos-sdk/x/upgrade/types"
 )
 
-// UpgradeName defines the on-chain upgrade name for the sample simap upgrade from v045 to v046.
+// UpgradeName defines the on-chain upgrade name for the sample SimApp upgrade
+// from v046 to v047.
 //
-// NOTE: This upgrade defines a reference implementation of what an upgrade could look like
-// when an application is migrating from Cosmos SDK version v0.45.x to v0.46.x.
-const UpgradeName = "v045-to-v046"
+// NOTE: This upgrade defines a reference implementation of what an upgrade
+// could look like when an application is migrating from Cosmos SDK version
+// v0.46.x to v0.47.x.
+const UpgradeName = "v046-to-v047"
 
 func (app SimApp) RegisterUpgradeHandlers() {
-	app.UpgradeKeeper.SetUpgradeHandler(UpgradeName,
-		func(ctx sdk.Context, plan upgradetypes.Plan, fromVM module.VersionMap) (module.VersionMap, error) {
+	baseAppLegacySS := app.ParamsKeeper.Subspace(baseapp.Paramspace).WithKeyTable(paramstypes.ConsensusParamsKeyTable())
+
+	app.UpgradeKeeper.SetUpgradeHandler(
+		UpgradeName,
+		func(ctx sdk.Context, _ upgradetypes.Plan, fromVM module.VersionMap) (module.VersionMap, error) {
+			if cp := baseapp.GetConsensusParams(ctx, baseAppLegacySS); cp != nil {
+				app.ConsensusParamsKeeper.Set(ctx, cp)
+			} else {
+				ctx.Logger().Info("warning: consensus parameters are undefined; skipping migration", "upgrade", UpgradeName)
+			}
```

Comment:
should we throw this in a helper function so its less boiler plate to copy over?

Reply:
I can think of something, yeah.